torchgeo/datasets/agb_live_woody_density.py

```diff
@@ -0,0 +1,178 @@
+
+    filename_glob = "*N_*E.*"
+    filename_regex = (
+        r"""^(?P<latitude>[0-9][0-9][A-Z])_(?P<longitude>[0-9][0-9][0-9][A-Z])*"""
+    )
+
+    def __init__(
+        self,
+        root: str = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        download: bool = False,
+        cache: bool = True,
+    ) -> None:
+        """Initialize a new Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            cache: if True, cache file handle to speed up repeated sampling
+
+        Raises:
+            FileNotFoundError: if no files are found in ``root``
+        """
+        self.root = root
+        self.download = download
+
+        self._verify()
+
+        super().__init__(root, crs, res, transforms, cache)
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if dataset is missing
+        """
+        # Check if the extracted files already exist
+        pathname = os.path.join(self.root, self.filename_glob)
+        if glob.glob(pathname):
+            return
+
+        # Check if the downloaded base file already exist
+        pathname = os.path.join(self.root, self.base_filename)
+        if glob.glob(pathname):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automaticaly download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        download_url(self.url, self.root, self.base_filename)
+
+    def _extract(self) -> None:
+        with open(os.path.join(self.root, self.base_filename), "r") as f:
+            content = json.load(f)
+
+        for item in content["features"]:
+            download_url(
+                item["properties"]["download"],
+                self.root,
+                item["properties"]["tile_id"] + ".tif",
+            )
+
+    def plot(  # type: ignore[override]
+        self,
+        sample: Dict[str, Tensor],
```

Comment:
```suggestion
        sample: Dict[str, Any],
```
See my comment on your other PR